### PR TITLE
Add API to get the syntax tree from a given context

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject as per the shell requirement. 

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/SyntaxTree.java [hunk 1]
```diff
@@ -64,6 +65,29 @@ public static SyntaxTree from(SyntaxTree oldTree, TextDocumentChange textDocumen
...
+    public static SyntaxTree fromStatements(TextDocument textDocument) {
+        return fromGivenContext(textDocument, null, ParserRuleContext.STATEMENTS);
+    }
+
+    public static SyntaxTree fromExpression(TextDocument textDocument) {
+        return fromGivenContext(textDocument, null, ParserRuleContext.EXPRESSION);
+    }
+
+    public static SyntaxTree fromGivenContext(TextDocument textDocument, String filePath, ParserRuleContext context) {
+        // TODO: Can get rid of overloaded methods once ParserRuleContext is exposed to outside.
```
Comment (lochana-chathura) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/SyntaxTree.java:85]: @rdulmina Once we have a proper place to expose parser internals, we can remove the loaded methods. For now, this is okay? WDYT?

Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/SyntaxTree.java:85]: We can also rename `fromGivenContext` to `fromContext`. Or a better name suggestion?

Comment (rdulmina) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/SyntaxTree.java:85]: +1 but the names doesn't make sense to me

Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/SyntaxTree.java]: Names updated as per offline discussion. Therefore resolving this.

**Review by rdulmina — APPROVED** [review]

**Pull request merged**